src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const updateState = [];
+  let cloneObj = clone(state);
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      cloneObj = Object.assign(cloneObj, action.extraData);
+      updateState.push(clone(cloneObj));
```

Comment:
Dont repeat yourself. 'updateState.push(...)' you can use only one time after all cases.